plugins/woocommerce/includes/shortcodes/class-wc-shortcode-checkout.php

```diff
@@ -183,12 +183,31 @@ private static function order_pay( $order_id ) {
 					current( $available_gateways )->set_current();
 				}
 
+				/**
+				 * Allows the text of the submit button on the Pay for Order page to be changed.
+				 *
+				 * @param string $text The text of the button.
+				 *
+				 * @since 3.0.2
+				 */
+				$order_button_text = apply_filters( 'woocommerce_pay_order_button_text', __( 'Pay for order', 'woocommerce' ) );
+
+				/**
+				 * Triggered right before the Pay for Order form, after validation of the order and customer.
+				 *
+				 * @param WC_Order $order             The order that is being paid for.
+				 * @param string   $order_button_text The text for the submit button.
+				 *
+				 * @since 6.6
+				 */
+				do_action( 'before_woocommerce_pay_form', $order, $order_button_text );
```

Comment:
Looks good. 

I think the same thing could be achieved with the existing `woocommerce_before_template_part` hook, but this is going to be more performant.

Quick question: though it might not be needed for your immediate purposes, would there be value in also making `$available_gateways` available to callbacks (for instance, if I wanted to conditionally add something like _"Pay using XYZ and receive bonus points")?_ Not a blocker though, as this could be added in the future without any back-compat concerns. Pre-emptively marking as approved, feel free to merge!

Reply:
> Quick question: though it might not be needed for your immediate purposes, would there be value in also making $available_gateways available to callbacks (for instance, if I wanted to conditionally add something like "Pay using XYZ and receive bonus points")? Not a blocker though, as this could be added in the future without any back-compat concerns. Pre-emptively marking as approved, feel free to merge!

This is an oversight on my side. My thought was that we (as "the gateway guys") already know which gateway we're handling. Added in https://github.com/woocommerce/woocommerce/pull/33078/commits/9b6331e88c81d502cf23f6db510ffa1a24288f31 now, no need to have other PRs to modify it in the future.